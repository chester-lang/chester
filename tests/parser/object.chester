%{
  boot.loader.systemd-boot.enable = true,
  boot.kernelModules = [ ],
  console = %{
    font = "Lat2-Terminus16",
    keyMap = "us",
  },
}