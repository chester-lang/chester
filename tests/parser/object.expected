Block(
  statements = Vector(),
  result = Some(
    value = ObjectExpr(
      clauses = Vector(
        ObjectExprClause(
          key = DotCall(
            expr = DotCall(
              expr = DotCall(
                expr = Identifier(name = "boot", meta = None),
                field = Identifier(name = "loader", meta = None),
                telescope = Vector(),
                meta = None
              ),
              field = Identifier(name = "systemd-boot", meta = None),
              telescope = Vector(),
              meta = None
            ),
            field = Identifier(name = "enable", meta = None),
            telescope = Vector(),
            meta = None
          ),
          value = Identifier(name = "true", meta = None)
        ),
        ObjectExprClause(
          key = DotCall(
            expr = Identifier(name = "boot", meta = None),
            field = Identifier(name = "kernelModules", meta = None),
            telescope = Vector(),
            meta = None
          ),
          value = ListExpr(terms = Vector(), meta = None)
        ),
        ObjectExprClause(
          key = Identifier(name = "console", meta = None),
          value = ObjectExpr(
            clauses = Vector(
              ObjectExprClause(
                key = Identifier(name = "font", meta = None),
                value = StringLiteral(value = "Lat2-Terminus16", meta = None)
              ),
              ObjectExprClause(key = Identifier(name = "keyMap", meta = None), value = StringLiteral(value = "us", meta = None))
            ),
            meta = None
          )
        )
      ),
      meta = None
    )
  ),
  meta = None
)